File: Python/Bindings/Phantasma.py
import requests
import json

class PhantasmaAPI:
    url = None 

    def __init__(self, url):
        self.url = url

    def __JSON_RPC(self, method, params):
        headers = {'content-type': 'application/json'}

        payload = {
                "method": method,
                "params": params,
                "jsonrpc": "2.0",
                "id": 1,
        }
        response = requests.post(self.url, data=json.dumps(payload), headers=headers).json()

        assert response["jsonrpc"]
        assert response["id"] == 1
        return response["result"]

{{#each methods}}
	def {{#camel-case Info.Name}}(self{{#each Info.Parameters}}, {{Name}}{{/each}}):	
		"""
		{{Info.Description}}
		{{#if Info.Parameters}}
		Args:{{#each Info.Parameters}}
			{{Name}}: {{Description}}{{/each}}
		{{#else}}
		Args:
			None
		{{/if}}
		"""
		params = [{{#each Info.Parameters}}{{Name}}{{#if !@last}}, {{/if}}{{/each}}];
		return self.__JSON_RPC("{{#camel-case Info.Name}}", params);	
	{{/each}}
	

File: JS/Bindings/phantasma.js
class PhantasmaAPI {

	const Opcodes = Object.freeze({
		{{#each opcodes}}{{#upper-case Key}}:   {{Value}},
		{{/each}}
	});

	constructor(host) {
		this.host = host;
	}

	JSONRPC(method, params, onSuccess, onError) {
		let message = {
			'jsonrpc': '2.0',
			'method': method,
			'params': params,
			'id': '1'
		};
		let json = JSON.stringify(message);
		console.log(json);
		
		let request = new XMLHttpRequest();

		request.open('POST', this.host, true);

		request.onreadystatechange = function() {
			console.log(this.readyState);
			if (this.readyState === 4) { 
				if (this.status === 200) {
					let data = JSON.parse(request.responseText);
					if (data.result) {
						onSuccess(data.result);
					}
					else {
						onError('invalid response');				
					}
				}
				else {
					onError('connection error');				
				}
			} else {
				return true;
			}		  			
		};

		request.onerror = function() {
			onError('internal error');
		};

		request.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
		request.setRequestHeader('Accept', 'application/json'); 
		request.send(json);
		console.log("sent");
	}

	convertDecimals(amount, decimals) {
		let mult = Math.pow(10, decimals);
		return amount / mult;
	}

	{{#each methods}}
	//{{Info.Description}}
	{{#camel-case Info.Name}}({{#each Info.Parameters}}{{Name}},{{/each}}onSuccess, onError)
	{
		let params = [{{#each Info.Parameters}}{{Name}}{{#if !@last}}, {{/if}}{{/each}}];
		this.JSONRPC('{{#camel-case Info.Name}}', params, onSuccess, onError);
	}
{{/each}}}